stl/inc/yvals_core.h

```diff
@@ -415,6 +411,23 @@
 #include <vcruntime.h>
 #include <xkeycheck.h> // The _HAS_CXX tags must be defined before including this.
 
+#define _STL_PRAGMA(PRAGMA)          _Pragma(#PRAGMA)
+#define _STL_PRAGMA_MESSAGE(MESSAGE) _STL_PRAGMA(message(MESSAGE))
+#define _EMIT_STL_MESSAGE(MESSAGE)   _STL_PRAGMA_MESSAGE(__FILE__ "(" _CRT_STRINGIZE(__LINE__) "): " MESSAGE)
+
+// clang-format off
+#define _EMIT_STL_WARNING(NUMBER, MESSAGE) \
+    _EMIT_STL_MESSAGE("warning STL" #NUMBER ": " MESSAGE) \
+    static_assert(1, "")
+#define _EMIT_STL_ERROR(NUMBER, MESSAGE) \
+    _EMIT_STL_MESSAGE("error STL" #NUMBER ": " MESSAGE) \
+    static_assert(0, "Error in C++ Standard Library usage.")

```

Comment:
`static_assert` contextually converts its first operand to `bool`, so it would be simpler to say `true` and `false` here instead of `1` and `0`. (That is, if we need a `bool`, we conventionally use a `bool` instead of an integer constant.)

Reply:
Note the comment a few lines above.

Reply:
Doesn't `static_assert(false)` make this code IFNDR?

Reply:
No, it's in a macro definition, so unless the macro gets expanded it won't be anything; otherwise, it's fully ill-formed.

Reply:
> Note the comment a few lines above.

`// clang-format off`? `// The _HAS_CXX tags must be defined before including this.`?

Reply:
... did I delete the comment before pushing...?